### PR TITLE
Replace dai by nst

### PR DESCRIPTION
**Pull request by sunbreak1211**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,17 +4,17 @@ Implementations of MakerDAO surplus auctions, triggered on `vow.flap`.
 
 ### Splitter
 
-Exposes a `kick` operation to be triggered periodically. Its logic withdraws `DAI` from the `vow` and splits it in two parts. The first part (`burn`) is sent to the underlying `flapper` contract to be processed by the burn engine. The second part (`WAD - burn`) is distributed as reward to a `farm` contract. The `kick` cadence is determined by the `hop` value.
+Exposes a `kick` operation to be triggered periodically. Its logic withdraws `NST` from the `vow` and splits it in two parts. The first part (`burn`) is sent to the underlying `flapper` contract to be processed by the burn engine. The second part (`WAD - burn`) is distributed as reward to a `farm` contract. The `kick` cadence is determined by the `hop` value.
```
Comment (telome) [inline README.md:7]: Should we use `vat.dai` here? 
```suggestion
Exposes a `kick` operation to be triggered periodically. Its logic withdraws `vat.dai` from the `vow`, converts it into `NST`, and splits it in two parts. The first part (`burn`) is sent to the underlying `flapper` contract to be processed by the burn engine. The second part (`WAD - burn`) is distributed as reward to a `farm` contract. The `kick` cadence is determined by the `hop` value.
```

Comment (telome) [reply README.md:7]: Maybe better to keep it simple, as you did.

Comment (sunbreak1211) [reply README.md:7]: Both seem correct to me. As it is true that initiates with a `vat.dai` move, but then exists to `NST` (as before it did to `DAI`). So I would probably slightly tend to leave it with `NST`.

File: deploy/FlapperInit.sol [hunk 2]
```diff
@@ -112,14 +112,14 @@ library FlapperInit {
...
         // Sanity checks
         require(flapper.spotter()  == address(dss.spotter),                       "Flapper spotter mismatch");
-        require(flapper.dai()      == cfg.dai,                                    "Flapper dai mismatch");
+        require(flapper.nst()      == cfg.nst,                                    "Flapper nst mismatch");
         require(flapper.pair()     == cfg.pair,                                   "Flapper pair mismatch");
         require(flapper.receiver() == dss.chainlog.getAddress("MCD_PAUSE_PROXY"), "Flapper receiver mismatch");
 
         PairLike pair = PairLike(flapper.pair());
-        (address pairDai, address pairGem) = pair.token0() == cfg.dai ? (pair.token0(), pair.token1())
+        (address pairDai, address pairGem) = pair.token0() == cfg.nst ? (pair.token0(), pair.token1())
```
Comment (telome) [inline deploy/FlapperInit.sol:120]: Some local vars, types and functions were not renamed:
- in `src`: `pairDai`, `reserveDai`, `DaiJoinLike` (in the Splitter).
- in `test`: `reserveDai`, `minimalDaiReserve`, `currentDaiForGem`, `UNIV2_LINK_DAI_PAIR`, `_getDaiToSell`, `uniV2GemForDai`, `uniV2DaiForGem`, and some test function names. 

Was that intentional to reduce the diff? I'd tend to at least fully do the renaming for the files in `src`.

File: deploy/FlapperInit.sol [hunk 3]
```diff
@@ -50,14 +50,14 @@ interface PairLike {
     function token1() external view returns (address);
 }
 
-interface DaiJoinLike {
-    function dai() external view returns (address);
+interface NstJoinLike {
```
Comment (telome) [inline deploy/FlapperInit.sol:53]: For consistency we should also rename `DaiJoinLike` > `NstJoinLike` in the Splitter (and SplitterMock).

File: README.md (hunk 1 above)
Comment (telome) [reply README.md:7]: Ok, let's keep it as-is, resolving.

**Review by telome — APPROVED** [review]
LGTM

**Review by oldchili — APPROVED** [review]
LGTM

**Pull request merged**